### PR TITLE
Prevent GetUpNext from returning episodes in progress

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

GetUpNext was returning episodes that are in progress, leading to this:

Screenshot_2020-12-01 Jellyfin

Its purpose should be to only return future episodes of a show that is already started, but not in progress episodes.

To achieve this, this gets the user data of the episode to be added to the list, checks if it's in progress and just returns null if it is.

**Issues**

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

**Review by BaronGreenback — APPROVED** [review]

File: Emby.Server.Implementations/TV/TVSeriesManager.cs [hunk 1]
```diff
@@ -192,7 +192,7 @@ private static string GetUniqueSeriesKey(Series series)
 
             Func<Episode> getEpisode = () =>
             {
-                return _libraryManager.GetItemList(new InternalItemsQuery(user)
+                var nextEpsiode = _libraryManager.GetItemList(new InternalItemsQuery(user)
```
Comment (crobibero) [inline Emby.Server.Implementations/TV/TVSeriesManager.cs:195]:
```suggestion
                var nextEpisode = _libraryManager.GetItemList(new InternalItemsQuery(user)
```

File: Emby.Server.Implementations/TV/TVSeriesManager.cs [hunk 2]
```diff
@@ -205,6 +205,15 @@ private static string GetUniqueSeriesKey(Series series)
                     MinSortName = lastWatchedEpisode?.SortName,
                     DtoOptions = dtoOptions
                 }).Cast<Episode>().FirstOrDefault();
+
+                var userData = _userDataManager.GetUserData(user, nextEpsiode);
```
Comment (cvium) [inline Emby.Server.Implementations/TV/TVSeriesManager.cs:209]: Possible null ref here. If `nextEpsiode` is null it will crash.

**Review by cvium — CHANGES_REQUESTED** [review]
Fix nullref and spelling

File: Emby.Server.Implementations/TV/TVSeriesManager.cs [hunk 3]
```diff
@@ -205,6 +205,17 @@ private static string GetUniqueSeriesKey(Series series)
                     MinSortName = lastWatchedEpisode?.SortName,
                     DtoOptions = dtoOptions
                 }).Cast<Episode>().FirstOrDefault();
+
+                if (!(nextEpisode is null)) {
```
Comment (cvium) [inline Emby.Server.Implementations/TV/TVSeriesManager.cs:209]:
```suggestion
                if (nextEpisode != null) {
```

**Review by cvium — APPROVED** [review]

**Pull request merged**